Deploy docs to a `gh-pages` branch

slifty opened the pull request:
This PR adds a workflow to publish documentation pages to `gh-pages` every time there is a merge to main.

Resolves #291

slifty commented on .github/workflows/deploy_docs.yml:
```diff
@@ -0,0 +1,31 @@
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "22"
```
We should use a `.node-version` file and reference it in our workflows, but I didn't want to cause trouble in this PR

slifty commented on .github/workflows/deploy_docs.yml:
```diff
@@ -0,0 +1,31 @@
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "18"
```
Until #215 is done I'm sticking with the version of node other workflows use

slifty commented:
For one cycle I added the pages push to this PR's workflow, and it did indeed push to gh-pages (https://github.com/PermanentOrg/stela/tree/gh-pages)

So it seems to work!

liam-lloyd commented on .github/workflows/test.yml:
```diff
@@ -46,3 +46,18 @@ jobs:
       - run: (cd stela; npm run test -w @stela/access_copy_attacher)
       - uses: codecov/codecov-action@v2
+
+  verify_docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "18"
+
+      - name: Install Redoc CLI
+        run: npm install -g @redocly/cli
+
+      - name: Generate HTML from OpenAPI spec
+        run: npx @redocly/cli build-docs ./packages/api/docs/src/api.yaml -o docs/index.html
```
I'm not sure that we need this; I think the `redocly lint` that we run as part of linting will catch anything that would make a build fail (though the docs don't declare that, it's just what I've seen so far).

slifty replied:
I'll take this out then!

slifty commented:
For one cycle I added the pages push to this PR's workflow, and it did indeed push to gh-pages (https://github.com/PermanentOrg/stela/tree/gh-pages)

So it seems to work!

I also updated our settings so that pages is serving from `gh-pages`
